### DISCUSSION TITLE
get k8s artifacts error: could not parse reference: docker-pullable

### DISCUSSION THREAD

**Discussion by AkikoOrenji** [aquasecurity/trivy#5685] [Bugs]
### Description

Trying to run a k8 audit just to see misconfigurations of the local cluster e.g. permissions, groups etc

### Desired Behavior

run trivy and get a report about kube

### Actual Behavior

FATAL   get k8s artifacts error: could not parse reference: docker-pullable://OUR-PRIVATE-DOCKER-REPO.ap-southeast-2.compute.internal:12376/cilium/cilium@sha256:somesha256sum

### Reproduction Steps

```bash
./trivy k8s --report=all --scanners config all
```

### Target

Kubernetes

### Scanner

Misconfiguration

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
2023-11-30T01:03:17.573Z        FATAL   get k8s artifacts error:
    github.com/aquasecurity/trivy/pkg/k8s/commands.namespaceRun
        /home/runner/work/trivy/trivy/pkg/k8s/commands/namespace.go:28
```

### Operating System

Redhat EL

### Version

```bash
Version: 0.47.0
Policy Bundle:
  Digest: sha256:1df8ade71efc830877ca3b1130f83e0c6368e3a45b0d4c0f0418955501644054
  DownloadedAt: 2023-11-30 00:31:12.648819818 +0000 UTC
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
@chen-keinan Can you please take a look?

**Comment by chen-keinan** [comment]
@AkikoOrenji Converting to issue #5685  for further investigation
